### ISSUE TITLE
Calculated column jinja templating fails in ORDER BY clause 

### ISSUE THREAD

**Issue by lilotter45** [apache/superset#29378]
### Bug description

When attempting to use a calculated column that uses jinja templating as the metric in a bar chart, the jinja placeholders are included in the `ORDER BY` clause of the executed query, rather than being replaced with rendered SQL string. The template renders correctly in the `SELECT` statement of the same query.

In this example, the calculated column is defined as 

```python
IIF(MeasureValue < 
    {% if filter_values('fresh_min_bcs')|length %} 
        {{ filter_values('fresh_min_bcs')[0] }} 
    {% else %} 
        2.75 
    {% endif %},
    1,
    0
)
```

The executed query (trimmed for brevity) is

```sql
WITH cteMeasure AS (SELECT ... FROM ...),
     __cte AS (SELECT ... )
SELECT TOP 10000 rpt.DateTrunc('QUARTER' , MeasureDate) AS [MeasureDate_Group],
           IIF(SortNum >= 3 , '3' , CAST(SortNum AS VARCHAR(2))) AS [SortGroup],
           sum(IIF(MeasureValue < 2.75, 1, 0)) AS [COUNT(Is Low BCS)]
FROM __cte
WHERE ...
GROUP BY ...
-- The template does not render correctly in the ORDER BY clause on the following line
ORDER BY sum(IIF(MeasureValue < {% if filter_values('fresh_min_bcs')|length %} {{ filter_values('fresh_min_bcs')[0] }} {% else %} 2.75 {% endif %}, 1, 0)) DESC;
```

### How to reproduce the bug

1. Create a dataset with a calcuated field that uses jinja templating
2. Create a Bar Chart based on the above dataset
3. Add the calculated column as a *METRIC*; the aggregration function does not matter
4. Click *UPDATE CHART*
5. An "Unexpected error" will be returned, citing "Incorrect syntax near '%'"; clicking on `...` and "View query" will show the template text rather than rendered text in the `ORDER BY` clause of the executed query

### Superset version

4.0.1

### Python version

3.10

### Node version

Not applicable

### Browser

Chrome

### Additional context

- Superset is installed via pip, in a virtual environment
- `ENABLE_TEMPLATE_PROCESSING` feature flag is enabled
- The dataset is a virtual dataset from a query against MS SQL Server 2019: `Microsoft SQL Server 2019 (RTM-CU27) (KB5037331) - 15.0.4375.4 (X64)  	Jun 10 2024 18:00:07  	Copyright (C) 2019 Microsoft Corporation 	Standard Edition (64-bit) on Windows Server 2019 Standard 10.0 <X64> (Build 17763: ) (Hypervisor) `

### Checklist

- [X] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [X] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [X] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by rusackas** [comment]
Are you still facing this issue? It's been quiet for a long time, and may be a better fit for a Help/Q&A Discussion.

**Comment by lilotter45** [comment]
Yes, I am still facing this issue. I actually just came across it via a web search while facing the same issue in another dataset.

I'm not clear why this would be a better fit for a discussion. Should the jinja template not be rendered consistently between the `SELECT`, `WHERE`, `GROUP BY`, and `ORDER BY` clauses?

Another example:

```
CASE
  WHEN '{% if filter_values('arrival_date_field')|length %}{{ filter_values('arrival_date_field')[0] }}{% else %}Birth Date{% endif %}' = 'Arrival Date' THEN 
    arrival_date 
  ELSE 
    birth_date 
  END
```

evaluates to `'Birth Date'` in the `SELECT` and `WHERE` clauses, but is interpreted as the raw string in the `ORDER BY` clause:

```
ORDER BY MAX(CASE
  WHEN '{% if filter_values('arrival_date_field')|length %}{{ filter_values('arrival_date_field')[0] }}{% else %}Birth Date{% endif %}' = 'Arrival Date' THEN 
    arrival_date 
  ELSE 
    birth_date 
  END) DESC
```

When manually replacing the jinja template in SQL Lab with the string it would evaluate to, the query executes successfully.

**Comment by rusackas** [comment]
I wonder if @Vitor-Avila or @villebro have any input here, as they're MUCH more versed with the ins and outs of Jinja than I am. @dosu might be able to help here as well, better than it did a year ago :)

**Comment by Vitor-Avila** [comment]
@rusackas I can reproduce this on my end, agree it seems like a bug. I'll see if it's an easy fix

**Comment by Hai6123** [comment]
I faced the same problem when trying to create a dynamic bar chart, but I accidentally came up with this _WORKAROUND_ by using jinja to create **Virtual Dataset**
About the context, my group by column is **"_ten_chi_nhanh_"**, my filter here is named **"_metrics2_"** with chosen value is **"_ADG_TB_"**, my code is kinda shit since it is in developing phase

This is the code for the  **Virtual Dataset**:

<img width="527" height="244" alt="Image" src="https://github.com/user-attachments/assets/79d64ba2-27e4-44c6-b0d9-b0724de8f0d8" />

This is my returned query:

<img width="777" height="419" alt="Image" src="https://github.com/user-attachments/assets/19b001c6-a435-4e57-9f7d-8c937e7d04c3" />

And this is the former query with this bug:

<img width="776" height="424" alt="Image" src="https://github.com/user-attachments/assets/23737c73-95ad-4763-b778-d99a857f6b12" />
